editor/scene_tree_editor.cpp

```diff
@@ -431,29 +431,21 @@ bool SceneTreeEditor::_add_nodes(Node *p_node, TreeItem *p_parent, bool p_scroll
 		}
 	}
 
-	bool scroll = false;
-
 	if (editor_selection) {
 		if (editor_selection->is_selected(p_node)) {
 			item->select(0);
-			scroll = p_scroll_to_selected;
 		}
 	}
 
 	if (selected == p_node) {
 		if (!editor_selection) {
 			item->select(0);
-			scroll = p_scroll_to_selected;
 		}
 		item->set_as_cursor(0);
 	}
 
-	bool keep = (filter.is_subsequence_ofn(String(p_node->get_name())));
```

Comment:
The old code I rememberd is only change L451 to:
![image](https://user-images.githubusercontent.com/17956756/183471761-db3f23f1-3c58-4ba2-80f7-3a1abed2f577.png)
(sorry I have only this screenshot for old commit, the code works in godot master this Feb 13th.)